core/src/main/java/tc/oc/pgm/PGMConfig.java

```diff
@@ -0,0 +1,549 @@
+
+    if (!config.getBoolean("restart.enabled", true)) {
+      config.set("restart.uptime", -1);
+      config.set("restart.match-limit", -1);
+    }
+    config.set("restart.memory", null);
+    config.set("restart.enabled", null);
+
+    renameKey(config, "start.countdown", "countdown.start");
+    renameKey(config, "start.huddle", "countdown.huddle");
+    if (!config.getBoolean("start.auto", true)) {
+      config.set("countdown.start", -1);
+    }
+    config.set("start", null);
+
+    renameKey(config, "cycle.countdown", "countdown.cycle");
+    if (!config.getBoolean("cycle.match-empty.enabled", true)) {
+      config.set("join.min-players", 0);
+    }
+    config.set("cycle", null);
+
+    renameKey(config, "join.mid-match", "join.anytime");
+    if (config.getBoolean("join.commit-players", false)
+        || config.getBoolean("teams.require-even")) {
+      config.set("join.queue", true);
+    }
+    config.set("join.commit-players", null);
+    renameKey(config, "join.capacity.enabled", "join.limit");
+    config.set("join.capacity", null);
+
+    renameKey(config, "teams.try-balance", "join.balance");
+    renameKey(config, "minimum-players", "join.min-players");
+
+    config.set("teams", null);
+    config.set("broadcast", null);
+    config.set("arrow-removal", null);
+    config.set("fishing", null);
+
+    renameKey(config, "scoreboard.show-proximity", "ui.proximity");
+    config.set("scoreboard", null);
+
+    renameKey(config, "wool.auto-refill", "gameplay.refill-wool");
+    config.set("wool", null);
+
+    renameKey(config, "player-list.enabled", "ui.tablist");
+    config.set("player-list", null);
+
+    if (config.getBoolean("motd.enabled", true)) {
+      final String format =
+          config
+              .getString("motd.format")
+              .replace("{state.color}", "{2}")
+              .replace("{map.name}", "{1}")
+              .replace("{map.version}", "1.0.0")
+              .replace("{state.name}", "Idle")
+              .replace("{state.name-lower}", "idle");
+      config.set("motd", format);
+    } else {
+      config.set("motd", "");
+    }
+
+    // priority:id:prefix
+    final List<String> groups = new ArrayList<>();
+
+    final ConfigurationSection section = config.getConfigurationSection("groups");
+    if (section != null) {
+      for (String key : section.getKeys(false)) {
+        final String id = section.getBoolean(key + ".op", false) ? "op" : key;
+        final int priority = section.getInt(key + ".priority", Integer.MAX_VALUE);
+        String prefix = null;
+        try {
+          prefix =
+              "&"
+                  + parseEnum(section.getString(key + ".prefix.color"), ChatColor.class).getChar()
+                  + section.getString(key + ".prefix.symbol");
+        } catch (TextException | NullPointerException e) {
+          // No-op
+        }
+
+        groups.add(String.format("%s|%s|%s", priority, id, prefix == null ? "" : prefix));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: priority (declared at line 230). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 241. Consider formatting this data with &#39;%d&#39; instead.